translator/trace/zipkin/zipkinv2_to_traces.go

```diff
@@ -0,0 +1,454 @@
+			return err
+		}
+
+		dropped, errDropped := strconv.ParseUint(parts[partCnt-1], 10, 32)
+		if errDropped != nil {
+			return errDropped
+		}
+		link.SetDroppedAttributesCount(uint32(dropped))
+	}
+	return nil
+}
+
+func populateSpanEvents(zspan *zipkinmodel.SpanModel, events pdata.SpanEventSlice) error {
+	events.Resize(len(zspan.Annotations))
+	for ix, anno := range zspan.Annotations {
+		event := events.At(ix)
+		startNano := anno.Timestamp.UnixNano()
+		event.SetTimestamp(pdata.TimestampUnixNano(startNano))
+
+		parts := strings.Split(anno.Value, "|")
+		partCnt := len(parts)
+		event.SetName(parts[0])
+		if partCnt < 3 {
+			continue
+		}
+
+		var jsonStr string
+		if partCnt == 3 {
+			jsonStr = parts[1]
+		} else {
+			jsonParts := parts[1 : partCnt-2]
+			jsonStr = strings.Join(jsonParts, "|")
+		}
+		var attrs map[string]interface{}
+		if err := json.Unmarshal([]byte(jsonStr), &attrs); err != nil {
+			return err
+		}
+		if err := jsonMapToAttributeMap(attrs, event.Attributes()); err != nil {
+			return err
+		}
+
+		dropped, errDropped := strconv.ParseUint(parts[partCnt-1], 10, 32)
+		if errDropped != nil {
+			return errDropped
+		}
+		event.SetDroppedAttributesCount(uint32(dropped))
+	}
+	return nil
+}
+
+func jsonMapToAttributeMap(attrs map[string]interface{}, dest pdata.AttributeMap) error {
+	for key, val := range attrs {
+		if s, ok := val.(string); ok {
+			dest.InsertString(key, s)
+		} else if i, ok := val.(int64); ok {
+			dest.InsertInt(key, i)
+		} else if d, ok := val.(float64); ok {
+			dest.InsertDouble(key, d)
+		} else if b, ok := val.(bool); ok {
+			dest.InsertBool(key, b)
+		}
+	}
+	return nil
+}
+
+func zTagsToInternalAttrs(zspan *zipkinmodel.SpanModel, tags map[string]string, dest pdata.AttributeMap) error {
+	parseErr := tagsToAttributeMap(tags, dest)
+	if zspan.LocalEndpoint != nil {
+		if zspan.LocalEndpoint.IPv4 != nil {
+			dest.InsertString(conventions.AttributeNetHostIP, zspan.LocalEndpoint.IPv4.String())
+		}
+		if zspan.LocalEndpoint.IPv6 != nil {
+			dest.InsertString(conventions.AttributeNetHostIP, zspan.LocalEndpoint.IPv6.String())
+		}
+		if zspan.LocalEndpoint.Port > 0 {
+			dest.UpsertInt(conventions.AttributeNetHostPort, int64(zspan.LocalEndpoint.Port))
+		}
+	}
+	if zspan.RemoteEndpoint != nil {
+		if zspan.RemoteEndpoint.ServiceName != "" {
+			dest.InsertString(conventions.AttributePeerService, zspan.RemoteEndpoint.ServiceName)
+		}
+		if zspan.RemoteEndpoint.IPv4 != nil {
+			dest.InsertString(conventions.AttributeNetPeerIP, zspan.RemoteEndpoint.IPv4.String())
+		}
+		if zspan.RemoteEndpoint.IPv6 != nil {
+			dest.InsertString(conventions.AttributeNetPeerIP, zspan.RemoteEndpoint.IPv6.String())
+		}
+		if zspan.RemoteEndpoint.Port > 0 {
+			dest.UpsertInt(conventions.AttributeNetPeerPort, int64(zspan.RemoteEndpoint.Port))
+		}
+	}
+	return parseErr
+}
+
+func tagsToAttributeMap(tags map[string]string, dest pdata.AttributeMap) error {
```

Comment:
This logic is different than the previous parsing logic. I'd appreciate if this remained unified with v1 also.

https://github.com/open-telemetry/opentelemetry-collector/blob/master/translator/trace/zipkin/zipkinv1_to_protospan.go#L253-L266

I don't think ints should need to be in the `intAttributes` to be parsed. 

Support for bools was dropped, it should be added back.

Reply:
Changed to use same logic